test/CMakeLists.txt

```diff
@@ -1,9 +1,27 @@
+include(FindPkgConfig)
+
 llvm_canonicalize_cmake_booleans(SPIRV_SKIP_CLANG_BUILD)
 llvm_canonicalize_cmake_booleans(SPIRV_SKIP_DEBUG_INFO_TESTS)
 
 # required by lit.site.cfg.py.in
 get_target_property(LLVM_SPIRV_DIR llvm-spirv BINARY_DIR)
 set(LLVM_SPIRV_TEST_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+
+# find spirv-val
+pkg_search_module(SPIRV_TOOLS SPIRV-Tools)
+if(SPIRV_TOOLS_FOUND)
+  find_program(SPIRV_TOOLS_SPIRV_VAL NAMES spirv-val PATHS ${SPIRV_TOOLS_PREFIX}/bin)
+  if(SPIRV_TOOLS_SPIRV_VAL)
+    get_filename_component(SPIRV_TOOLS_BINDIR "${SPIRV_TOOLS_SPIRV_VAL}" DIRECTORY)
+    set(SPIRV_TOOLS_SPIRV_VAL_FOUND True)
+  endif()
+endif()
+
+if(NOT SPIRV_TOOLS_SPIRV_VAL)
+  message(WARNING "spirv-val not found! Correctness of spirv can't be verified by provided tests")
```

Comment:
spirv-val does not check that the translation was actually correct, only that the result conforms to the SPIR-V spec, so I'd rephrase the second sentence as:
"SPIR-V generated for test suite will not be validated."